storage/storage.go

```diff
@@ -291,6 +295,27 @@ func (s *Storage) RemoveAggregation(id string) error {
 	return nil
 }
 
+func AggregationFromMap(m map[string]string) (job.Aggregation, error) {
+	var err error
+	aggr := job.Aggregation{}
+	for k, v := range m {
+		switch k {
+		case "ID":
+			aggr.ID = v
+		case "Limit":
+			aggr.Limit, err = strconv.Atoi(v)
+			if err != nil {
+				return aggr, fmt.Errorf("Could not decode struct from map: %v", err)
+			}
+		case "Proxy":
+			aggr.Proxy = v
+		default:
+			return aggr, fmt.Errorf("Field %s with value %s was not found in Aggregarion struct", k, v)
+		}
+	}
+	return aggr, nil
+}
```

Comment:
Just a thought: instead of iterating m, could we do something like this instead?

```golang
func AggregationFromMap(m map[string]string) (job.Aggregation, error) {
    return NewAggregation(m["id"], m["limit"], m["proxy"])
}
```

Of course, this way we don't handle the case where the map contained other, irrelevant fields, but I'm not convinced we should handle this case at all.

Reply:
TBH since this is the way we do it in the Job struct, I think we should leave it as is for now.

I fully agree that it is bloated to iterate over the map, but both of those functions need refactoring to a more dynamic logic, so that we do not need to add code everytime we change fields.

Reply:
Sure, we could do this in a subsequent PR since it's out-of-scope currently.